src/backend/distributed/executor/executor.c

```diff
@@ -622,16 +629,34 @@ DistributedStatementRequiresRollback(DistributedPlan *distributedPlan)
  * distributed execution. In particular, it releases connections and
  * clears their state.
  */
-void
+static void
 FinishDistributedExecution(DistributedExecution *execution)
 {
-	DistributedPlan *distributedPlan = execution->plan;
+	UnsetCitusNoticeLevel();
+
+	if (execution->operation != CMD_SELECT)
```

Comment:
Would be good to put this in a function.

Reply:
Bump

Reply:
Sorry, missed that. Do you mean moving these two lines into a function like
```C
MarkAsModifyingXacts()
```
etc?

Reply:
I meant using the `DistributedExecutionModifiesDatabase` function instead of `!= CMD_SELECT`